hnn_core/network.py

```diff
@@ -429,7 +477,7 @@ def _record_spikes(self):
         # agnostic to type of source, will sort that out later
         for gid in self._gid_list:
             if pc.gid_exists(gid):
-                pc.spike_record(gid, self.spiketimes, self.spikegids)
+                pc.spike_record(gid, self.spikes.times, self.spikes.gids)
```

Comment:
how about `spikes.record(pc)`? That way you can keep everything (`times`, `gids` etc) as simple objects, e.g., numpy array and only in this method you instantiate `h.Vector()` ?

Reply:
actually never mind this comment, I'm not so sure of this